bot/connector-alcmeon/src/main/kotlin/AlcmeonConnectorProvider.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2017/2021 e-voyageurs technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.tock.bot.connector.alcmeon
+
+import ai.tock.bot.connector.Connector
+import ai.tock.bot.connector.ConnectorConfiguration
+import ai.tock.bot.connector.ConnectorMessage
+import ai.tock.bot.connector.ConnectorProvider
+import ai.tock.bot.connector.ConnectorType
+import ai.tock.bot.connector.ConnectorTypeConfiguration
+import ai.tock.bot.connector.ConnectorTypeConfigurationField
+import ai.tock.bot.connector.messenger.messengerConnectorType
+import ai.tock.bot.connector.whatsapp.whatsAppConnectorType
+import ai.tock.shared.resourceAsString
+import com.fasterxml.jackson.annotation.JsonProperty
+import kotlin.reflect.KClass
+
+private const val NAME_FIELD = "name"
+private const val DESCRIPTION_FIELD = "description"
+private const val BACKENDS_FIELD = "backends"
+private const val EXIT_NAME_FIELD = "exit_name"
+private const val EXIT_DESCRIPTION_FIELD = "exit_description"
+
+internal object AlcmeonConnectorProvider : ConnectorProvider {
+
+    override val connectorType: ConnectorType get() = alcmeonConnectorType
+
+    override fun connector(connectorConfiguration: ConnectorConfiguration): Connector {
+        with(connectorConfiguration) {
+            val authorisationHandler = AlcmeonAuthorisationHandler()
+
+            return AlcmeonConnector(
+                connectorId,
+                path,
+                AlcmeonSubBotDescription(
+                    name = parameters[NAME_FIELD] ?: "SNCF Connect",
+                    description = parameters[DESCRIPTION_FIELD] ?: "Bot SNCF Connect",
+                    backends = parameters[BACKENDS_FIELD]?.split(",")?.map { it.trim() } ?: listOf(
+                        "whatsapp",
+                        "facebook"
+                    ),
+                    exits = listOf(
+                        AlcmeonSubBotExit(
+                            parameters[EXIT_NAME_FIELD] ?: "alcmeon",
+                            parameters[EXIT_DESCRIPTION_FIELD] ?: "Back to alcmeon customer service",
+                        )
+                    )),
```

Comment:
May be easier to only fill one field description in configuration with the complete description

Reply:
LGTM